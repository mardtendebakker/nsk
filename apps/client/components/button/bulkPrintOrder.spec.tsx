import { fireEvent, render, waitFor } from '@testing-library/react';
import BulkPrintOrder from './bulkPrintOrder';

const mockAxios = {
  call: jest.fn(() => Promise.resolve({ data: 'data' })),
};

jest.mock('../../hooks/useAxios', () => () => mockAxios);
jest.mock('../../utils/blob', () => ({
  openBlob: jest.fn(),
}));

describe('bulkPrint', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  describe('tooltip mode', () => {
    it('matches snapshot', () => {
      const { asFragment } = render(<BulkPrintOrder onPerforming={() => {}} type="purchase" ids={['1']} />);
      expect(asFragment()).toMatchSnapshot();
    });
    it('handles click', async () => {
      const onPerforming = jest.fn();
      const { getByRole } = render(<BulkPrintOrder onPerforming={onPerforming} type="purchase" ids={['1']} />);
      await waitFor(() => { fireEvent.click(getByRole('button')); });

      expect(mockAxios.call).toHaveBeenCalled();
      expect(onPerforming).toHaveBeenCalledWith(true);
      expect(onPerforming).toHaveBeenCalledWith(false);
    });
    it('disables click', async () => {
      const onPerforming = jest.fn();
      const { getByRole } = render(<BulkPrintOrder onPerforming={onPerforming} type="purchase" ids={['1']} disabled />);
      await waitFor(() => { fireEvent.click(getByRole('button')); });

      expect(mockAxios.call).not.toHaveBeenCalled();
      expect(onPerforming).not.toHaveBeenCalled();
    });
  });
});
