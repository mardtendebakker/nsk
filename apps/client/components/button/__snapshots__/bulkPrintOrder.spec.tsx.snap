// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bulkPrint tooltip mode matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall css-1k23hlb-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    print
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-uym7v6-MuiSvgIcon-root"
      data-testid="ChevronRightIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
      />
    </svg>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
